### PR TITLE
Gitlabci improvements

### PR DESCRIPTION
**Pull request by atodorov**
This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -30,6 +30,8 @@ RPM:
     EXTRA_REPO_PATH_SEGMENT: "gitlab/"
   script:
     - sh "schutzbot/mockbuild.sh"
+  after_script:
+    - schutzbot/update_github_status.sh update
```
Comment (jrusz) [inline .gitlab-ci.yml:34]: This should be already included from `extends: .terraform` no? Or does it just override the after_script from that?

Comment (atodorov) [reply .gitlab-ci.yml:34]: it overrides the value from the `.terraform` leaving only this script to be executed and not all of the others.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -88,6 +88,10 @@ Base:
           # see COMPOSER-918
           # - aws/rhel-8.4-aarch64
         INTERNAL_NETWORK: ["true"]
+  artifacts:
+    paths:
```
Comment (ondrejbudai) [inline .gitlab-ci.yml:92]: What's the reason behind uploading the repo file in the base tests? I would expect it in the RPM build stage.

Comment (ondrejbudai) [reply .gitlab-ci.yml:92]: Oh, the journal is now not saved for base tests.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -30,6 +30,8 @@ RPM:
     EXTRA_REPO_PATH_SEGMENT: "gitlab/"
   script:
     - sh "schutzbot/mockbuild.sh"
+  after_script:
```
Comment (ondrejbudai) [inline .gitlab-ci.yml:33]: The removal of `save_journal` causes following lines in the gitlab log:

```
WARNING: journal-log: no matching files            
ERROR: No files to upload
```

**Review by ondrejbudai — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (atodorov) [reply .gitlab-ci.yml:92]: > What's the reason behind uploading the repo file in the base tests? I would expect it in the RPM build stage.

The repo file is created via `setup_repo()` in `deploy.sh`, which isn't executed during rpmbuild.

**Review by ondrejbudai — APPROVED** [review]
Looks good, thanks!

**Pull request merged**